search/geocoder.cpp

```diff
@@ -735,6 +776,24 @@ void Geocoder::ForEachCountry(vector<shared_ptr<MwmInfo>> const & infos, TFn &&
   }
 }
 
+void Geocoder::MatchCategories(BaseContext & ctx)
+{
+  auto emit = [&](uint64_t bit) {
+    auto const featureId = base::asserted_cast<uint32_t>(bit);
+    Model::Type type;
+    if (!GetTypeInGeocoding(ctx, featureId, type))
+      return;
+
+    EmitResult(ctx, m_context->GetId(), featureId, type, TokenRange(0, 1), nullptr /* geoParts */);
+  };
+
+  // By now there's only one token and zero prefix tokens.
```

Comment:
Where the intersection with pivot is done?

Reply:
https://github.com/mapsme/omim/pull/7213/files/2846b625ed4d5644b76f8ece3a3d30226f978db7#diff-7e907361d64c0be8c2f5b2053636ef1eR542

Reply:
Although, additional intersection in the non-viewport case won't hurt. Working on it.

Reply:
done